ir/memory.h

```diff
@@ -91,16 +93,74 @@ class Pointer {
 };
 
 
+// A data structure that represents a byte.
```

Comment:
This whole Byte class is an implementation detail of the memory. I don't see a need to expose it to users.
I suggest moving it completely to memory.cpp instead.

Reply:
Makes sense, moved it to memory.cpp